src/library/scala/collection/immutable/LazyList.scala

```diff
@@ -590,7 +590,11 @@ final class LazyList[+A] private(private[this] var lazyState: () => LazyList.Sta
 
   @inline private def lazySlidingImpl(size: Int, step: Int): Iterator[LazyList[A]] =
     if (knownIsEmpty) Iterator.empty
-    else Iterator.empty ++ slidingImpl(size, step, size - step max 0) // concat with empty iterator so that `slidingImpl` is lazy
+    else Iterator.empty ++ startSliding(size, step) // concat with empty iterator so that it's lazy
+
+  private def startSliding(size: Int, step: Int): Iterator[LazyList[A]] =
```

Comment:
This method is called exactly once by `lazySlidingImpl`.  Can you just inline it there?

Reply:
probably. will look into it later today. it was mostly a concern about ugliness/readability

Reply:
I might also just convert the whole thing to an explicit `Iterator` rather than doing all this `concat`ing